shared/src/main/scala/org/adridadou/openlaw/parser/template/variableTypes/LinkType.scala

```diff
@@ -0,0 +1,60 @@
+package org.adridadou.openlaw.parser.template.variableTypes
+
+import cats.implicits._
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.parser.decode
+import org.adridadou.openlaw.{OpenlawNativeValue, OpenlawString, OpenlawValue}
+import org.adridadou.openlaw.parser.template.formatters.Formatter
+import org.adridadou.openlaw.parser.template._
+import org.adridadou.openlaw.result.{Failure, FailureException, Result, Success}
+
+case class LinkInfo(label: String, url:String) extends OpenlawNativeValue
+
+case object LinkType extends VariableType(name = "Link") {
+
+  private implicit val enc: Encoder[LinkInfo] = deriveEncoder[LinkInfo]
+  private implicit val dec: Decoder[LinkInfo] = deriveDecoder[LinkInfo]
+
+  override def cast(value: String, executionResult: TemplateExecutionResult): Result[LinkInfo] = decode[LinkInfo](value).leftMap(FailureException(_))
+
+  override def internalFormat(value: OpenlawValue): Result[String] = VariableType.convert[OpenlawString](value)
+
+  override def defaultFormatter: Formatter = new LinkFormatter
+
+  override def getTypeClass: Class[LinkInfo] = classOf[LinkInfo]
+
+  override def checkTypeName(nameToCheck: String): Boolean = Seq("Link").exists(_.equalsIgnoreCase(nameToCheck))
+
+  override def construct(constructorParams: Parameter, executionResult: TemplateExecutionResult): Result[Option[LinkInfo]] = {
+    constructorParams match {
+      case Parameters(seq) =>
+        val map = seq.toMap
+        (for {
```

Comment:
No need to change this, but just a note that since `sequence` is a `traverse` with the identity function, a slightly shorter alternative would be to do the traverse with the transforming function up front, e.g.

```
for {
  label <- map.get("label").traverse(getOneValueConstant)
  url <- map.get("url").traverse(getOneValueConstant)
} yield (label, url) mapN { LinkInfo(_, _) }
```

(the `mapN` just being a shorthand for applying the `LinkInfo` constructor to both `label` and `url` options)

Reply:
thank you! shorter sounds a bit better in this case :) Will change!